pynucastro/networks/rate_collection.py

```diff
@@ -86,62 +86,90 @@ def _skip_xp(n, p, r):
     return False
 
 
-class Composition:
+class Composition(collections.UserDict):
     """a composition holds the mass fractions of the nuclei in a network
     -- useful for evaluating the rates
 
     """
     def __init__(self, nuclei, small=1.e-16):
         """nuclei is an iterable of the nuclei in the network"""
         try:
-            self.X = {Nucleus.cast(k): small for k in nuclei}
+            super().__init__({Nucleus.cast(k): small for k in nuclei})
         except TypeError:
             raise ValueError("must supply an iterable of Nucleus objects or strings") from None
 
-    def __str__(self):
-        ostr = ""
-        for k in self.X:
-            ostr += f"  X({k}) : {self.X[k]}\n"
-        return ostr
```

Comment:
Can you add this `__str__()` back, for nicer output in notebooks and such? The current `__repr__()` is fine as-is though